code/backend/batch/utilities/helpers/azure_computer_vision_client.py

```diff
@@ -0,0 +1,75 @@
+import logging
+from typing import List
+from urllib.parse import urljoin
+from azure.identity import DefaultAzureCredential, get_bearer_token_provider
+
+import requests
+from requests import Response
+
+from .env_helper import EnvHelper
+
+logger = logging.getLogger(__name__)
+
+
+class AzureComputerVisionClient:
+
+    __TOKEN_SCOPE = "https://cognitiveservices.azure.com/.default"
+    __VECTORIZE_IMAGE_PATH = "computervision/retrieval:vectorizeImage"
+    __VECTORIZE_IMAGE_API_VERSION = "2024-02-01"
+    __VECTORIZE_IMAGE_MODEL_VERSION = "2023-04-15"
+    __RESPONSE_VECTOR_KEY = "vector"
+
+    def __init__(self, env_helper: EnvHelper) -> None:
+        self.computer_vision_host = env_helper.AZURE_COMPUTER_VISION_ENDPOINT
+        self.computer_vision_key = env_helper.AZURE_COMPUTER_VISION_KEY
+        self.use_keys = env_helper.is_auth_type_keys()
+
+    def vectorize_image(self, image_url: str) -> List[float]:
+        logger.info("Making call to computer vision to vectorize image")
+        response = self.__make_request(image_url)
```

Comment:
do we want to have this call in try-catch ? what happens if the call fails ?

Reply:
We've put the request in a try/except block, this should catch errors where we do not receive a proper HTTP response

As part of this, we've configured a timeout